src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1498,11 +1509,16 @@
   -->
   <AbstractNode Name="BoundTupleExpression" Base="BoundExpression">
     <Field Name="Arguments" Type="ImmutableArray&lt;BoundExpression&gt;"/>
+
+    <Field Name="ArgumentNamesOpt" Type="ImmutableArray&lt;string&gt;" Null="allow"/>
+
+    <!-- Which argument names were inferred (as opposed to explicitly provided)? -->
+    <Field Name="InferredNamesOpt" Type="ImmutableArray&lt;bool&gt;" Null="allow"/>
   </AbstractNode>
 
   <!-- 
   Tuple literals can convert to the target type.
-  Once converted to a target type, they cannot be target-typed again. 
+  Once converted to a target type, they cannot be target-typed again.
   The tuple literal is one which has not been converted to a target type.
   -->
```

Comment:
Comment should mention that this node does not survive past initial binding. Same thing for unconverted switch expression.
Maybe we could even enforce that.

Reply:
It does survive initial binding, as a converted literal points to the unconverted literal.  That makes a bit of a mess in our bound trees (or should i say bound dag), but that is what it is today.

---
In reply to: [301753825](https://github.com/dotnet/roslyn/pull/37052#discussion_r301753825) [](ancestors = 301753825)